lib/addons/backstage/index.ts

```diff
@@ -0,0 +1,103 @@
+import { Construct } from 'constructs';
+import { HelmAddOn, HelmAddOnUserProps } from "../helm-addon";
+import { dependable, setPath } from "../../utils";
+import { ClusterInfo, Values } from "../../spi";
+import { ICertificate } from "aws-cdk-lib/aws-certificatemanager";
+
+/**
+ * User provided options for the Helm Chart
+ */
+export interface BackstageAddOnProps extends HelmAddOnUserProps {
+    version?: string,
+    name?: string, 
+    createNamespace?: boolean,
+    namespace?: string,
+    subdomain: string,
+    certificateResourceName: string,
+    imageRegistry: string,
+    imageRepository: string,
+    imageTag?: string,
+    baseUrl: string,
+    postgresHost: string,
+    postgresPort?: number,
+    postgresUser?: string,
+    postgresPassword: string
+}
+
+/**
+ * Default props to be used when creating the Helm chart
+ */
+const defaultProps = {
+  name: "blueprints-backstage-addon",
+  namespace: "backstage",
+  chart: "backstage",
+  version: "0.17.0",
+  release: "backstage",
+  repository:  "https://backstage.github.io/charts",
+  imageTag: "latest",
+  postgresPort: 5432,
+  postgresUser: "postgres",
+  values: {}
+};
+
+/**
+ * Main class to instantiate the Helm chart
+ */
+export class BackstageAddOn extends HelmAddOn {
+
+  readonly options: BackstageAddOnProps;
+
+  constructor(props?: BackstageAddOnProps) {
+    super({...defaultProps, ...props});
+    this.options = this.props as BackstageAddOnProps;
+  }
+  
+  @dependable('AwsLoadBalancerControllerAddOn')
+  deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+    let values: Values = populateValues(clusterInfo, this.options);
+    const chart = this.addHelmChart(clusterInfo, values);
+
+    return Promise.resolve(chart);
+  }
+}
+
+/**
+ * populateValues populates the appropriate values used to customize the Helm chart
+ * @param helmOptions User provided values to customize the chart
+ */
+function populateValues(clusterInfo: ClusterInfo, helmOptions: BackstageAddOnProps): Values {
+  const values = helmOptions.values ?? {};
+  
+  const annotations = {
```

Comment:
"alb" in the annotations below = AWS Load Balancer controller; AwsLoadBalancerControllerAddOn is set as a dependency of the deploy, so the user would know it is a requirement; the other popular ingress controller option I could see is NGINX; the related addon depends on AWS Load Balancer controller anyway though (as per https://aws-quickstart.github.io/cdk-eks-blueprints/addons/nginx/), hence I didnt make the ingress controller configurable. opinionated decision. If we wanted to make it a more open, we would probably need to have a set of acceptable ingress controllers, the user would need to add the related add-on, pass the selected type as parameter and we would need to have annotations per ingress controller (and test them). We would also need to insure the ingress controller classname is installed (see the related selection in setPath "ingress.className" below ), otherwise the process will fail. It is certainly something useful, but I feel it could be worked on at a later stage, without affecting the functionality of a first working version of Backstage. Any other approach you would recommend?

Reply:
"alb" in the annotations below = AWS Load Balancer controller; AwsLoadBalancerControllerAddOn is set as a dependency of the deploy, so the user would know it is a requirement; the other popular ingress controller option I could see is NGINX; the related addon depends on AWS Load Balancer controller anyway though (as per https://aws-quickstart.github.io/cdk-eks-blueprints/addons/nginx/), hence I didnt make the ingress controller configurable. opinionated decision. If we wanted to make it a more open, we would probably need to have a set of acceptable ingress controllers, the user would need to add the related add-on, pass the selected type as parameter and we would need to have annotations per ingress controller (and test them). We would also need to insure the ingress controller classname is installed (see the related selection in setPath "ingress.className" below ), otherwise the process will fail. It is certainly something useful, but I feel it could be worked on at a later stage, without affecting the functionality of a first working version of Backstage. Any other approach you would recommend?